### PR TITLE
Added apm-infra correlation docs and a link to them from the troubleshooting page

### PR DESCRIPTION
**Pull request by harnitsignalfx**

### CODE REVIEW

**Comment by rmfitzpatrick** [comment]
Thanks for all this @harnitsignalfx.  Mind taking a look @tmorrisonsfx?

File: docs/troubleshooting.md [hunk 1]
```diff
@@ -169,3 +169,13 @@ $ curl -X POST localhost:9411/api/v2/spans -H'Content-Type: application/json' -d
...
+
+## APM - Infrastructure Correlation
+
+### Setup the APM Service - Infrastructure correlation metrics
+
+How do you setup the collector to capture and send the relevant data to show
+correlated infrastructure metrics in the APM service dashboards?
+
+Follow the steps outlined [here](apm-infra-correlation.md) and if you are still running into
+issues then reach out to [support](#gathering-support-information)
```
Comment (tmorrisonsfx) [inline docs/troubleshooting.md:181]: A few suggested changes:

- Change title to "Set up the APM service - infrastructure correlation metrics".
- Change "setup" to "set up".
- Change the last sentence to "Follow the steps outlined here. If you are not able to set up the APM service after following the instructions, contact support."
- Change "here" to the name of the page you are linking to.

File: docs/apm-infra-correlation.md [hunk 2]
```diff
@@ -0,0 +1,295 @@
+# How to Enable APM-Infra Correlation 
+
+In the APM Service Dashboards, there are charts that indicate the health of
+the underlying infrastructure. The default configuration of Splunk OpenTelemetry 
+Connector automatically configures this for you, but when specifying a custom 
+configuration it is important to keep the following in mind.
+
```
Comment (tmorrisonsfx) [inline docs/apm-infra-correlation.md:7]: Suggested change:

"The APM service dashboards include charts that indicate the health of the underlying infrastructure. The default configuration of Splunk OpenTelemetry Connector automatically configures this for you, but when specifying a custom configuration, it is important to keep the following in mind."

Also, when you say "configures this for you," are you referring to the dashboards?

File: docs/apm-infra-correlation.md [hunk 3]
```diff
@@ -0,0 +1,295 @@
+# How to Enable APM-Infra Correlation
```
Comment (tmorrisonsfx) [inline docs/apm-infra-correlation.md:1]: Suggested title change: "Enable APM-Infra Correlation"

File: docs/apm-infra-correlation.md [hunk 4]
```diff
@@ -0,0 +1,295 @@
+# How to Enable APM-Infra Correlation 
+
+In the APM Service Dashboards, there are charts that indicate the health of
+the underlying infrastructure. The default configuration of Splunk OpenTelemetry 
+Connector automatically configures this for you, but when specifying a custom 
+configuration it is important to keep the following in mind.
+
+
+
+## Configuration to enable APM-Infra Correlation from a standalone VM (Agent)
```
Comment (tmorrisonsfx) [inline docs/apm-infra-correlation.md:10]: I would change "to get your infrastructure data to show up," but I'm not quite sure how to keep the technical accuracy. For example, the user has to do something (enable those things in the config) to get the data to show up, which is different from saying something like "to view your infrastructure data". 

Also, I would change the list to a table.

File: docs/apm-infra-correlation.md (hunk 2 above)
Comment (harnitsignalfx) [reply docs/apm-infra-correlation.md:7]: ty @tmorrisonsfx ! "configures this for you" - this was a suggestion by another reviewer, the connector's default configuration already contains the changes that we mention here.

File: docs/apm-infra-correlation.md (hunk 3 above)
Comment (harnitsignalfx) [reply docs/apm-infra-correlation.md:1]: ack

File: docs/apm-infra-correlation.md (hunk 4 above)
Comment (harnitsignalfx) [reply docs/apm-infra-correlation.md:10]: Thats fair, I'll change this to "to view your infrastructure data". 

My only hesitancy with changing to a table is that one column will be extremely long compared to the rest. List is certainly cleaner. Thoughts?

File: docs/apm-infra-correlation.md [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+    - override enabled
+    
+    <em>This processor will enable a unique `host.name` value to be set for metrics 
+    and traces. The `host.name` is determined by either the ec2 host name, or the 
+    system host name.</em>
+   
+
+-  `resource/add_environment` processor (optional)
+    
+    <em>This processor inserts a `deployment.environment` span tag to all the spans. The APM
```
Comment (tmorrisonsfx) [inline docs/apm-infra-correlation.md:49]: Suggested changes:

- "... to all spans."
- "... span tag to be set correctly." 
- "You should configure this span tag in the instrumentation, but if that is not an option, you can use this processor ..."

File: docs/apm-infra-correlation.md [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+    the relevant APM service charts.</em>
+
+    - `correlation` flag (enabled by default)
+
+    - `signalfx` exporter must be enabled for metrics AND traces pipeline (example below)
+
+    <em>The `correlation` flag of the `signalfx` exporter will allow the 
+    collector to make relevant API calls to correlate your spans with the 
+    infrastructure metrics, but to do this the `signalfx` exporter must be placed 
+    in the traces pipeline. This flag is enabled by default, so no specific configuration
```
Comment (tmorrisonsfx) [inline docs/apm-infra-correlation.md:33]: Suggested changes:

- "... , but to do this, the `signalfx` exporter ..."
- "This flag is enabled by default, so there is no specific configuration needed."
- Change "fine tune" to "adjust" or "modify".
- Change "here" to the name of the page you are linking to.

File: docs/troubleshooting.md (hunk 1 above)
Comment (harnitsignalfx) [reply docs/troubleshooting.md:181]: @tmorrisonsfx for the last suggestion - 
I'm not sure how to construct that sentence without "here".
Follow the steps outlined <Enable APM Correlation> ?

File: docs/apm-infra-correlation.md (hunk 5 above)
Comment (harnitsignalfx) [reply docs/apm-infra-correlation.md:49]: ack

File: docs/apm-infra-correlation.md (hunk 6 above)
Comment (harnitsignalfx) [reply docs/apm-infra-correlation.md:33]: ack to the first 3. 
Similar problems with "here", not sure how to construct an appropriate sentence with the page name.

**Comment by tmorrisonsfx** [comment]
> Thanks for all this @harnitsignalfx. Mind taking a look @tmorrisonsfx?

Hi @harnitsignalfx. I have completed my review. Let me know if you have any questions. Thanks!

File: docs/apm-infra-correlation.md [hunk 7]
```diff
@@ -0,0 +1,294 @@
...
+    metrics:
+      receivers: [hostmetrics]
+      processors: [memory_limiter, batch, resourcedetection]
+      exporters: [signalfx]
+```     
+
+## Configuration to enable APM-Infra Correlation from an Agent -> Gateway
+If you need to run the opentelemetry collector as an Agent and a Gateway, then you should follow
+the steps listed below for running the collector in each mode (Agent vs Gateway). 
+
```
[the remark this answers is not in the document]
Comment (harnitsignalfx) [reply docs/apm-infra-correlation.md:105]: ty!

File: docs/apm-infra-correlation.md [hunk 8]
```diff
@@ -0,0 +1,294 @@
...
+      receivers: [jaeger, zipkin]
+      processors: [memory_limiter, batch, resourcedetection, resource/add_environment]
+      exporters: [sapm, signalfx]
+    metrics:
+      receivers: [hostmetrics]
+      processors: [memory_limiter, batch, resourcedetection]
+      exporters: [signalfx]
+```     
+
+## Configuration to enable APM-Infra Correlation from an Agent -> Gateway
```
[the remark this answers is not in the document]
Comment (harnitsignalfx) [reply docs/apm-infra-correlation.md:102]: ack, ty!

File: docs/apm-infra-correlation.md [hunk 9]
```diff
@@ -0,0 +1,294 @@
...
+      processors: [memory_limiter, batch, resourcedetection, resource/add_environment]
+      exporters: [otlp, signalfx]
+    metrics:
+      receivers: [hostmetrics]
+      processors: [memory_limiter, batch, resourcedetection]
+      exporters: [otlp]
+``` 
+
+### Gateway
+On the side of the Gateway, you'd need the relevant receivers to match the exporters from
```
[the remark this answers is not in the document]
Comment (harnitsignalfx) [reply docs/apm-infra-correlation.md:200]: ack, ty

File: docs/apm-infra-correlation.md [hunk 10]
```diff
@@ -0,0 +1,294 @@
...
+the Agent. In addition, you'd need the following specific changes.
+
+- `http_forwarder` extension
+    - `egress` endpoint
+
+    <em>The `http_forwarder` listens on port 6060 and sends all the REST API calls directly
```
[the remark this answers is not in the document]
Comment (harnitsignalfx) [reply docs/apm-infra-correlation.md:206]: ack, ty

**Review by dmitryax — APPROVED** [review]

**Review by rmfitzpatrick — APPROVED** [review]
Thanks again!

**Comment by harnitsignalfx** [comment]
Sorry @rmfitzpatrick - didn't mean to re-request a review. Not sure how to cancel it. Please ignore !

**Pull request merged**